osprey_ui/src/lib/copyText.ts

```diff
@@ -0,0 +1,20 @@
+// the clipboard api is not available when using a non-secure host (aside from localhost)
+// this can be pretty common for some setups that are running i.e. on tailscale, so ideally
+// we don't want to break things when a user tries to copy text somewhere. here we use a
+// fallback when either that api is not available or `secureContext` is false
+// see: https://developer.mozilla.org/en-US/docs/Web/API/Navigator/clipboard
+export async function copyText(text: string): Promise<void> {
+  if (navigator.clipboard && window.isSecureContext) {
+    return navigator.clipboard.writeText(text);
+  } else {
+    // create a textarea that gets appended to the body, then focus+copy text
+    const body = document.body;
+    const textarea = document.createElement('textarea');
+    body.appendChild(textarea);
+    textarea.value = text;
+    textarea.focus();
+    textarea.select();
+    document.execCommand('copy');
+    textarea.remove();
```

Comment:
```suggestion
    try {
      document.execCommand('copy');
    }
    catch (err) {
      console.log(`copyText: fallback failed: ${err}`);
    }
    finally {
      textarea.remove();
    }
```

Since the `copy` command is [formally deprecated](https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand), consider wrapping it in `try` to ensure that the cleanup textarea removal fires even if the browser vendors decide to break this down the road. (Or with the various new browsers in development, who knows if they'll choose to implement deprecated functionality or not.)

Reply:
that makes sense! i changed this to actually log an error instead of a generic info log so that it's more obvious (there might be an alert system somewhere in this codebase that im unaware of that could get used too...)